deep-learning/src/main/python/synapse/ml/dl/sentence_embedding_transformer.py

```diff
@@ -0,0 +1,498 @@
+# Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+class SuppressLogging:
+    def __init__(self):
+        self._original_stderr = None
+
+    def start(self):
+        """Start suppressing logging by redirecting sys.stderr to /dev/null."""
+        if self._original_stderr is None:
+            self._original_stderr = sys.stderr
+            sys.stderr = open('/dev/null', 'w')
+
+    def stop(self):
+        """Stop suppressing logging and restore sys.stderr."""
+        if self._original_stderr is not None:
+            sys.stderr.close()
+            sys.stderr = self._original_stderr
+            self._original_stderr = None
+
+
+# Import necessary libraries
+import numpy as np
+import torch
+import pyspark.sql.functions as F
+import tensorrt as trt
+import logging
+import warnings
+import sys
+import datetime
+import pytz
+from tqdm import tqdm, trange
+from numpy import ndarray
+from torch import Tensor
+from typing import List, Union
+
+import model_navigator as nav
+from sentence_transformers import SentenceTransformer
+from sentence_transformers.util import batch_to_device
+from pyspark.ml.functions import predict_batch_udf
+from faker import Faker
+
+from pyspark.ml import Transformer
+from pyspark.ml.param.shared import HasInputCol, HasOutputCol, Param, Params
+from pyspark.sql.functions import col, struct, rand
+from pyspark.sql.types import (
+    StructType,
+    StructField,
+    IntegerType,
+    StringType,
+    ArrayType,
+    FloatType,
+)
+
+class EmbeddingTransformer(Transformer, HasInputCol, HasOutputCol):
```

Comment:
nit: HuggingFaceSentenceEmbedder

Also name the file HuggingFaceSentenceEmbedder.py